compiler/rustc_errors/src/emitter.rs

```diff
@@ -533,13 +533,13 @@ impl Emitter for HumanEmitter {
 /// An emitter that does nothing when emitting a non-fatal diagnostic.
 /// Fatal diagnostics are forwarded to `fatal_emitter` to avoid silent
 /// failures of rustc, as witnessed e.g. in issue #89358.
-pub struct SilentEmitter {
+pub struct FatalEmitter {
```

Comment:
The new name makes it seem like it will upgrade all diagnostics to fatal diagnostics to me.

Reply:
It could be FatalOnlyEmitter.